Better debug warp screen

garrettjoecox opened the pull request:
Extending some of the work done in #1221 to put a list of entrances together, this makes the debug screen a bit more useful.

- Main list has been shortened
- Removed opt and cutsceneIndex options
- Added entrance and daytime options

Before:
<img width="634" alt="Screen Shot 2022-09-14 at 10 18 42 PM" src="https://user-images.githubusercontent.com/7316699/190306099-7a3dbed7-752b-49de-96d9-8525c083af5e.png">

After:
<img width="638" alt="Screen Shot 2022-09-14 at 10 18 32 PM" src="https://user-images.githubusercontent.com/7316699/190306092-cae6414f-6942-4014-b4c0-09973a1d70f1.png">

Demo:


https://user-images.githubusercontent.com/7316699/190306211-a89130a5-8250-4127-ae1b-199760788148.mov

garrettjoecox committed:
Better debug warp screen

PurpleHato commented:
Great change but I think we should conserve the Cinematic index, it was very useful for testing and debugging each cutscenes of the game, it's also something I use literally a lot

garrettjoecox commented:
> Great change but I think we should conserve the Cinematic index, it was very useful for testing and debugging each cutscenes of the game, it's also something I use literally a lot

Oddly enough I haven't really had a use for the cutscene index once yet, but when I do play around with it I notice it's mostly broken (cutscenes duplicated, or sometimes not related to the scene selected) and very often crashes.

Worth noting though this is toggle-able option on the debug menu.

briaguya0 commented:
since the vanilla map select menu is still available and the current cutscene select can cause crashes, i think this PR should be good as is. a good future improvement could be adding cutscene select to the improved warp screen (with logic preventing loading broken cutscenes that cause crashes)

@PurpleHato thoughts?

PurpleHato commented:
> since the vanilla map select menu is still available and the current cutscene select can cause crashes, i think this PR should be good as is. a good future improvement could be adding cutscene select to the improved warp screen (with logic preventing loading broken cutscenes that cause crashes)
> 
> @PurpleHato thoughts?

Since it's toggleable, it's okay, and for the crashes for some cutscenes in vanilla debug mode,  I think it would need a massive test of each maps and scenario since there's lots of unused cutscenes too (even if irrelevent) and I think we would end up with too many conditions for that, so I guess there's no need to somehow prevent them, unless you want that to be super non-dev friendly who can't accept crashes when using an invalid cutscene id?

briaguya0 commented:
>  think it would need a massive test of each maps and scenario since there's lots of unused cutscenes too

i agree, which is why i think it could be it's own issue

> I think we would end up with too many conditions for that

probably would end up going the other way with it, knowing which cutscenes are valid for each scene and only having those as options

>  unless you want that to be super non-dev friendly 

i've definitely accidentally hit a button to set a crashing cutscene id before, i'd be a huge fan of that not being possible

another idea could be making cutscene select toggleable on the new map select menu, but even with that i'd still want to make an issue for "crash proof" cutscene select in the new map select menu

PurpleHato commented:
Sounds like a plan that will be it's own PR in the future and as a follow up of this PR then :D

aMannus commented:
Just want to mention https://github.com/HarbourMasters/Shipwright/pull/1221 also exists. They can co-exist just fine, just wanted to add it in case it becomes relevant.

garrettjoecox committed:
Rename and reorganize some warps

garrettjoecox commented:
Just did a pass on every warp cleaning up some ones that didn't make sense to be there and adjusted names and positions of others(previously the names and organization were directly from #1221). Also added `Warps`, `Shops`, and `Debug` sections at the end. This is good to go whenever 👍

briaguya0 reviewed (approved):
let's make an issue for adding crash-proof cutscene selection to this. but that's a future feature, for now let's :shipit:

LuigiXHero commented:
Might be dumb to bring up but the gray for text isn't very legible and would prefer it to either stay yellow or be a different bright color

garrettjoecox committed:
Persist scene selection to cvar & correctly wrap entrance selection

garrettjoecox committed:
Add grottos

garrettjoecox commented:
Made a few tweaks including adding grottos and making the grey text a bit brighter. Can we get this merged in? cc @dcvz

dcvz commented:
> Made a few tweaks including adding grottos and making the grey text a bit brighter. Can we get this merged in? cc @dcvz 

I'll check out the PR later today :)
